scalafix-cli/src/main/scala/scalafix/internal/v1/MainOps.scala

```diff
@@ -321,32 +322,44 @@ object MainOps {
       args: ValidatedArgs,
       input: Input,
       file: AbsolutePath
-  ): PatchInternals.ResultWithContext = {
-    val tree = LazyValue.later { () =>
-      args.parse(input).get: Tree
-    }
-    val doc = SyntacticDocument(input, tree, args.diffDisable, args.config)
-    if (args.rules.isSemantic) {
-      val relpath = file.toRelative(args.sourceroot)
-      val sdoc = SemanticDocument.fromPath(
-        doc,
-        relpath,
-        args.classLoader,
-        args.symtab,
-        () => compileWithGlobal(args, doc)
-      )
-      val result =
-        args.rules.semanticPatch(sdoc, args.args.autoSuppressLinterErrors)
-      assertFreshSemanticDB(
-        input,
-        file,
-        result.fixed,
-        sdoc.internal.textDocument
-      )
-      result
-    } else {
-      args.rules.syntacticPatch(doc, args.args.autoSuppressLinterErrors)
-    }
+  ): PatchInternals.ResultWithContext =
+    try {
+      val tree = LazyValue.later { () =>
+        args.parse(input).get: Tree
+      }
+      val doc = SyntacticDocument(input, tree, args.diffDisable, args.config)
+      if (args.rules.isSemantic) {
+        val relpath = file.toRelative(args.sourceroot)
+        val sdoc = SemanticDocument.fromPath(
+          doc,
+          relpath,
+          args.classLoader,
+          args.symtab,
+          () => compileWithGlobal(args, doc)
+        )
+        val result =
+          args.rules.semanticPatch(sdoc, args.args.autoSuppressLinterErrors)
+        assertFreshSemanticDB(
+          input,
+          file,
+          result.fixed,
+          sdoc.internal.textDocument
+        )
+        result
+      } else {
+        args.rules.syntacticPatch(doc, args.args.autoSuppressLinterErrors)
+      }
+    } finally shutdownCompiler(args.global)
```

Comment:
actually I guess the finally block could be scoped to where it's actually used - around the tunk calling `compileWithGlobal`?

Reply:
You are right, it's again on the wrong place .. we need it elsewhere, where the function is called. we don't want to open a new gloabl for each file.